cats/models.py

```diff
@@ -0,0 +1,99 @@
+"""
+Models used in app
+"""
+import datetime
+
+from django.core.exceptions import ValidationError
+from django.db import models
+
+STATUS_CHOICES = [
+    ('0', 'No status'),
+    ('1', 'Pending'),
+    ('2', 'Actual'),
+    ('3', 'Finished'),
+    ('4', 'Cancelled'),
+]
+
+
+class TimeStamped(models.Model):
+    """TimeStamp using when creating new object"""
+    created = models.DateTimeField(auto_now_add=True)
+    modified = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        abstract = True
+
+
+class CatQuerySet(models.QuerySet):
+    def filter_by_dates(self, rental_date, return_date):
+        """
+        Own method in QuerySet to filter out cats rented in given dates.
+
+        Returns a list of cats available between given dates.
+        """
+        rentals_list = Rental.objects.filter(
+            rental_date__gte=rental_date,
+            return_date__lte=return_date
+        )
+        rented_cats_id_list = rentals_list.values_list("cat")
+        available_cats = self.exclude(id__in=rented_cats_id_list)
+        return available_cats
```

Comment:
It should be possible to do something like that, but please test it
```suggestion
class CatQuerySet(models.QuerySet):
    def filter_available_between_dates(self, rental_date, return_date):
        """
        Own method in QuerySet to filter out cats rented in given dates.

        Returns a list of cats available between given dates.
        """
        return self.exclude(rentals__rental_date__gte=rental_date, rentals__return_date__lte=return_date)
```

Reply:
What with situation like this cat is rent between 1.1.22 and 31.1.22 and rental_date and return_date are 15.1.22 and 15.2.22 ?

Reply:
That's CarQuerySet. I want this method to filter Cats, not Rentals, so I won't agree with that change.

I'll only correct name of the method and make it's body little more compact.

Reply:
Sorry, I've just tested it out 😛
So true, your 1 line will do work here